src/service/search/mod.rs

```diff
@@ -389,11 +391,15 @@ pub async fn search_partition(
     if partitions.is_empty() {
         partitions.push([req.start_time, req.end_time]);
     }
-    if let Some((field, sort)) = sql.order_by.first() {
-        if field == &ts_column.unwrap() && !sort {
+
+    // We need to reverse partitions if query is ASC order
+    if let Some((field, order_by)) = sql.order_by.first() {
+        if field == &ts_column.unwrap() && order_by == &OrderBy::Asc {
+            resp.order_by = OrderBy::Asc;
             partitions.reverse();
         }
     }
+
```

Comment:
@hengfeiyang we should add the order in response, or let the FE figure it out?

Reply:
added, previous line.